Sources/CoreMetrics/SystemMetrics.swift

```diff
@@ -0,0 +1,322 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Metrics API open source project
+//
+// Copyright (c) 2018-2020 Apple Inc. and the Swift Metrics API project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift Metrics API project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Dispatch
+
+#if os(Linux)
+import Glibc
+#endif
+
+extension MetricsSystem {
+    /// `bootstrapSystemMetrics` is an one-time configuration function which globally enables system level metrics.
+    /// `bootstrapSystemMetrics` can be only called once, unless cancelled usung `cancelSystemMetrics`, calling it more
+    /// than once without cancelling will lead to undefined behaviour, most likely a crash.
+    /// There is no need to call `bootstrapSystemMetrics` if `SystemMetricsOptions` were provided to `bootstrap`.
+    ///
+    /// - parameters:
+    ///     - options: Options to configure `SystemMetricsHandler`
+    public static func bootstrapWithSystemMetrics(_ factory: MetricsFactory, options: SystemMetrics.Options) {
+        let factory = SystemMetricsFactory(factory: factory, options: options)
+        self.bootstrap(factory)
+        self.systemMetrics!.poll()
+    }
+    
+    /// Cancels the collection of system metrics. Calling this unlocks `bootstrapSystemMetrics` so it can be
+    /// called again.
+    public static func cancelSystemMetrics() {
+        self.systemMetrics?.cancelSystemMetrics()
+    }
+    
+    internal class SystemMetricsFactory: MetricsFactory {
+        fileprivate let queue = DispatchQueue(label: "com.apple.CoreMetrics.SystemMetricsHandler", qos: .background)
+        fileprivate let timeInterval: DispatchTimeInterval
+        fileprivate let systemMetricsType: SystemMetricsProvider.Type
+        fileprivate let labels: SystemMetrics.Labels
+        fileprivate var task: DispatchWorkItem?
```

Comment:
do we need a lock around task?

Reply:
I think in all cases (but I will double check) the `MetricsSystem` Lock is used, so there is no need in this specific context. But a good one to re-check

Reply:
I think we are assigning self.task in several places which may be unsafe?

Reply:
There are 2 places where we assign task. One is in the `init`, the other in the `deinit`. You could potentially have a case where the `deinit` is called at the exact same time as the next task execution, but I don't see that happening that quickly.

Wrapping in a lock shouldn't be too much work though, so I'm open to both keeping it this way, as well as locking it :)

Reply:
ah okay, I was under the impression there were more spots but maybe that got removed with cancel :+1: